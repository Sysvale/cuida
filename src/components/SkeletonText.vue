<template>
	<div
		class="skeleton__container"
		:style="computedStyle"
	/>
</template>

<script>
export default {
	props: {
		width: {
			type: String,
			required: false,
			default: '20',
		},

		height: {
			type: String,
			required: false,
			default: '16',
		},
	},

	computed: {
		computedStyle() {
			return  {
				'--width': `${this.width}px`,
				'--height': `${this.height}px`,
			};
		},
	},
}
</script>

<style lang="scss" scoped>
	@import '../assets/sass/tokens.scss';

	.skeleton__container {
		position: relative;
		overflow: hidden;
		cursor: wait;
		height: var(--height);
		width: var(--width);
		background-color: $n-30;
		border-radius: $border-radius-lil;
	}

	.skeleton__container:after {
		content: "";
		position: absolute;
		top: 0;
		right: 0;
		bottom: 0;
		left: 0;
		z-index: 0;
		background: linear-gradient(90deg, transparent, $n-20, transparent);
		animation: loading-animation 1.75s linear infinite;
	}

	@keyframes loading-animation {
		from {
			transform: translateX(-100%);
		}

		to {
			transform: translateX(100%);
		}
	}
</style>