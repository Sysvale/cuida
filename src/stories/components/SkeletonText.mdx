import { Meta, Story, Props, Controls, Canvas, Preview } from '@storybook/blocks';
import SkeletonText from '../../components/SkeletonText.vue';
import * as SkeletonTextStories from './SkeletonText.stories';

<Meta of={SkeletonTextStories} />

export const Template = (args) => ({
	components: { CdsSkeletonText: SkeletonText },
	setup() {
		return { args };
	},
	template: /*html*/ `
		<cds-skeleton-text
			v-bind="args" 
		/>
	`,
});

# Skeleton Text

### Skeleton texts permitem exibir um estado de carregamento para vários tipos de componentes

### enquanto seus dados estão sendo buscados.

***

<br />

## Quando usar:

* Informar ao usuário que as informações do componentes estão sendo carregadas.
* O conteúdo a ser carregado for dinâmico.

<br />

## Quando não usar:

* O conteúdo a ser carregado for estático.

## Observações

* O Skeleton text deve ser ajustado de acordo com o componente a ser utilizado.

<br />

## Preview

<Canvas>
  <Story of={SkeletonTextStories.SkeletonText} />
</Canvas>

<Controls />
