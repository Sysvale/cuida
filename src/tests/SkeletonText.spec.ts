import { describe, test, expect } from 'vitest';
import SkeletonText from '../components/SkeletonText.vue';
import { mount } from '@vue/test-utils';

describe('SkeletonText', () => {
	test('renders correctly', async () => {
		const wrapper = mount(SkeletonText, {
			props: {
				width: '300',
				height: '20',
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
