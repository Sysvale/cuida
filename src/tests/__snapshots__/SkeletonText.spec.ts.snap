// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkeletonText > renders correctly 1`] = `"<div data-v-58efc24d="" class="skeleton__container" style="--width: 300px; --height: 20px;"></div>"`;
